### PR TITLE
W-17282518 Add coverage 100 continue header  

### PR DESCRIPTION
**Pull request by aronee2008**

### CODE REVIEW

File: functional-tests/http/src/test/java/org/mule/test/http/functional/requester/HttpRequestExpect100ContinueHeaderTestCase.java [hunk 1]
```diff
@@ -65,4 +71,18 @@ private void doHandleRequestResponseIn(String flow) throws Exception {
     stopServer();
   }
 
+  @Test
+  public void testExpectContinueWhenServerTimesOut() throws Exception {
+    startExpectContinueTimeoutServer(persistentConnection);
+    final HttpRequestAttributes reqAttributes = mock(HttpRequestAttributes.class);
+
+    try {
+      flowRunner(REQUEST_FLOW_NAME_WITH_TIMEOUT).withAttributes(reqAttributes).withPayload(TEST_MESSAGE).run();
```
Comment (eze210) [inline functional-tests/http/src/test/java/org/mule/test/http/functional/requester/HttpRequestExpect100ContinueHeaderTestCase.java:80]: Use assertThrows() instead

File: functional-tests/http/src/test/java/org/mule/test/http/functional/requester/HttpRequestExpect100ContinueHeaderTestCase.java [hunk 2]
```diff
@@ -65,4 +71,18 @@ private void doHandleRequestResponseIn(String flow) throws Exception {
     stopServer();
   }
 
+  @Test
```
Comment (eze210) [inline functional-tests/http/src/test/java/org/mule/test/http/functional/requester/HttpRequestExpect100ContinueHeaderTestCase.java:74]: Add Issue Allure annotation

**Review by eze210 — CHANGES_REQUESTED** [review]
The new test looks fine, but didn't you want to add more corner cases?
Also, can these tests be done in MUnit?

File: functional-tests/http/src/test/java/org/mule/test/http/functional/requester/ntlm/HttpRequestExpect100ContinueHeaderNtlmAuthTestCase.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runners.Parameterized;
+
+import static org.mule.runtime.http.api.HttpHeaders.Names.AUTHORIZATION;
+import static org.mule.runtime.http.api.HttpHeaders.Names.WWW_AUTHENTICATE;
+
+import org.mule.test.runner.RunnerDelegateTo;
+
+@Issue("W-17282518")
```
Comment (aronee2008) [inline functional-tests/http/src/test/java/org/mule/test/http/functional/requester/ntlm/HttpRequestExpect100ContinueHeaderNtlmAuthTestCase.java:31]: If we are extending a class and using that to execute the assertions, do we add the allure annotation on top of the class ?

Comment (eze210) [reply functional-tests/http/src/test/java/org/mule/test/http/functional/requester/ntlm/HttpRequestExpect100ContinueHeaderNtlmAuthTestCase.java:31]: If all of the children classes will be related with this issue, putting the annotation in the parent class makes sense. In this case, it seems to be the case, so this annotation is ok

File: functional-tests/http/src/test/resources/http-request-expect-success-header-config.xml [hunk 4]
```diff
@@ -29,4 +31,15 @@
         </http:request>
     </flow>
 
+    <flow name="requestFlowWithLowTimeout">
+        <http:request config-ref="requestConfig" method="POST" path="/" responseTimeout="1">
+            <http:headers>
+                #[{'Expect' : '100-continue'}]
+            </http:headers>
+            <http:response-validator>
+                <http:success-status-code-validator values="0..599" />
```
Comment (eze210) [inline functional-tests/http/src/test/resources/http-request-expect-success-header-config.xml:40]: Why are we accepting status codes >=400 as success?

**Comment by eze210** [comment]
Why didn't you write these tests in MUnit directly?

**Comment by aronee2008** [comment]
I have created a new task to track the effort needed to migrate these tests to MUnit - https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE000026k7VTYAY/view
